python_modules/dagster/dagster/_core/definitions/time_window_partitions.py

```diff
@@ -71,19 +71,31 @@
 from .partition_key_range import PartitionKeyRange
 
 
-def is_ambiguous_time(dt: datetime):
-    """Returns if a datetime is ambiguous due to DST transitions."""
-    tz = check.not_none(dt.tzinfo)
+def is_second_ambiguous_time(dt: datetime, tz: str):
```

Comment:
tz_name or tz_string might be clearer, the `if tzinfo.tzname(None) != tz:` line broke my brain a little bit